planning/scenario_planning/lane_driving/motion_planning/surround_obstacle_checker/src/node.cpp

```diff
@@ -180,41 +169,56 @@ void SurroundObstacleCheckerNode::insertStopVelocity(
 }
 
 bool SurroundObstacleCheckerNode::getPose(
-  const std::string & source, const std::string & target, const ros::Time & time,
-  geometry_msgs::Pose & pose)
+  const std::string & source, const std::string & target, const rclcpp::Time & time,
+  geometry_msgs::msg::Pose & pose)
 {
   try {
     // get transform from source to target
-    geometry_msgs::TransformStamped ros_src2tgt;
-    ros_src2tgt = tf_buffer_.lookupTransform(source, target, time, ros::Duration(0.1));
-    // convert geometry_msgs::Transform to geometry_msgs::Pose
+    auto tf_future = tf_buffer_.waitForTransform(
```

Comment:
This is an example where `lookupTransform` was replaced by `waitForTransform`.

Reply:
I believe we decided not to use waitForTransform if we are asking for the newest transform and just use lookupTransform without timeout, correct?

Reply:
Yes, good point. I removed it.